Automattic/jetpack
# Align properly My Jetpack sections. Re organize heading components.

The final goal of this PR is align properly the plan and connection section titles. For this purpose, I've moved some components from `packages/my-jetpack` to `js-packages/components`, and consume these components from `packages/my-jetpack` and `js-packages/connection`.

These new components are entirely based on `Text component`, just adding margins to the box model in order to implement the given design:

<img width="736" alt="image" src="https://user-images.githubusercontent.com/77539/160397673-e52cf869-eeb2-4282-8c3b-4c2bf574a6a0.png">

Component | Prop | Typography | Example
------|------|------|------
`<H2>` | weight `bold` (default) | Headline Medium | `<H2>Hello, Medium world</H2>`
`<H3>` | weight `bold` (default) | Headline Small | `<H3>Hello, Small world</H3>`
`<H3>` | weight `regular` | Headline Small Regular | `<H3 weight="regular">Hello, Small and Regular world</H3>`
`<Title>` | size `medium` (default) | Title Medium | `<Title>Hello, Title world</Title>`
`<Title>` | size `small` | Title Small | `<Title size="small">Hello, Title small world</Title>`

Fixes https://github.com/Automattic/jetpack/issues/23627

#### Changes proposed in this Pull Request:

* Align properly My Jetpack sections. Re organize heading components.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Go to My Jetpack overview
* Inspect the element styles
* Confirm the titles of the plans and connect sections are aligned, same styles.
<img width="1143" alt="image" src="https://user-images.githubusercontent.com/77539/160398057-bb8eff31-d8b3-4cb2-baec-c14be3b34b09.png">

* Use Storybook
* Inspect the new `H2`, `H3` and `Title` components.
* Play with the weight (H2) and size (Title) props.


https://user-images.githubusercontent.com/77539/160471988-d014deb6-4a80-4d38-8185-92ef469f83ee.mov
